### ISSUE TITLE
go.mod requires go1.11, but os.ProcessState.ExitCode requires go1.12

### ISSUE THREAD

**Issue by andrewarchi** [git-lfs/git-lfs#4179]
**Describe the bug**
When building git-lfs from source with go1.11, the build fails because os.ProcessState.ExitCode is undefined (see https://github.com/git-lfs/git-lfs/issues/3945). Go added ExitCode to os.ProcessState in go1.12 ([release notes](https://golang.org/doc/go1.12#os)), however the git-lfs go.mod states a minimum requirement of go1.11.

Listing go1.11 is misleading and should be changed to 1.12. Additionally, it would be nice to note a minimum requirement on go1.12 in the README notes for installing [from source](https://github.com/git-lfs/git-lfs#from-source).

**To Reproduce**
1. Install go1.11
2. Build git-lfs
3. Observe build failures as described in https://github.com/git-lfs/git-lfs/issues/3945

**Expected behavior**
- go.mod should depend on `go 1.12` instead of `go 1.11`.
- README should note go1.12 minimum requirement in section on installing from source.

**System environment**
- FreeBSD 8, amd64
- `go version go1.11.4 freebsd/amd64`

**Output of `git lfs env`**
git-lfs did not build, thus this information is not available.

**Additional context**
We bootstrapped a patched go1.11.4 from source as Go no longer supports FreeBSD 8. Had it been known that git-lfs requires go1.12, we would have bootstrapped go1.12.17 instead.

Alternatively, if you want to continue supporting go1.11, [git/git.go:688](https://github.com/git-lfs/git-lfs/blob/43a83115f3d5b13e8432ccb9d7ac0d1880ff6d46/git/git.go#L688) could be changed to remove the call to `ExitCode`:
```diff
-if e, ok := err.(*exec.ExitError); ok && e.ProcessState.ExitCode() == 128 {
+if e, ok := err.(*exec.ExitError); ok && e.ProcessState.Sys().(syscall.WaitStatus).ExitStatus() == 128 {
```

**Comment by chrisd8088** [comment]
Hey, thanks for describing this and I'm sorry you're having trouble.

Per the changes in #3947 we only support the latest version of Go, but want to try to avoid breaking compatibility with older versions if possible; however, in #3959 we more or less fixed on Go 1.12 as the minimum version still supported.  (The conversation in #3945, which you referenced, also touches on this.)

We could adopt the `syscall.WaitStatus` technique you mention (also described in the answer to a Stack Overflow [question](https://stackoverflow.com/questions/10385551/get-exit-code-go)) as it sounds like it should work on Windows as well as Unix (Linux/macOS).  But we should, in that case, handle a failure to convert the `ProcessState.Sys()` value to `syscall.WaitStatus` too, as well as add a Go 1.11 build to our CI suite so we don't accidentally break 1.11 support in the future.

I proposed two PRs, one which just bumps `go.mod`'s minimum version (#4182) and the other which instead takes your suggestion for using `syscall.WaitStatus` and adds a Go 1.11 CI build as well (#4183).  We should choose one of these alternatives, I think; I'll leave it to the primary @git-lfs/core folks to consider which they prefer.

**Comment by andrewarchi** [comment]
Thanks, @chrisd8088.
